### PR TITLE
server: support MVCC-get HTTP API for clustered index

### PR DESCRIPTION
**Pull request by tangenta**
<!-- Thank you for contributing to TiDB!

PR Title Format:
1. pkg [, pkg2, pkg3]: what's changed
2. *: what's changed

-->

### What problem does this PR solve?

Problem Summary:
```sql
mysql> show create table t;
+-------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                                                                                                                               |
+-------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| t     | CREATE TABLE `t` (
  `a` varchar(255) NOT NULL,
  `b` int(11) DEFAULT NULL,
  `c` int(11) NOT NULL,
  PRIMARY KEY (`c`,`a`) /*T![clustered_index] CLUSTERED */
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin |
+-------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)

```

For now, we can't query the MVCC information from

```
curl http://127.0.0.1:10080/mvcc/key/test/t/{1-aaa}
```

because everything following the table name is parsed as `int64`:

```console
strconv.ParseInt: parsing "1-aaa": invalid syntax%
```

### What is changed and how it works?

This PR supports querying clustered index tables through MVCC HTTP API.

Here is an example usage:
```bash
curl http://127.0.0.1:10080/mvcc/key/test/t\?a\=aaaaa\&c\=22\&decode=1
```
```console
{
 "data": {
  "423122732719013888": [
   {
    "a": "aaaaa"
   },
   {
    "b": "11"
   }
  ]
 },
 "info": {
  "writes": [
   {
    "start_ts": 423122732719013888,
    "commit_ts": 423122732719800320,
    "short_value": "gAACAAAAAQIFAAYAYWFhYWEL"
   }
  ],
  "values": [
   {
    "start_ts": 423122732719013888,
    "value": "gAACAAAAAQIFAAYAYWFhYWEL"
   }
  ]
 },
 "key": "7480000000000000475F72038000000000000016016161616161000000FC"
}% 
```


### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:

### Check List <!--REMOVE the items that are not applicable-->

Tests <!-- At least one of them must be included. -->

- Unit test
- Integration test

Side effects

NA

### Release note <!-- bugfixes or new feature need a release note -->

- No release note

### CODE REVIEW

File: server/http_handler.go [hunk 1]
```diff
@@ -156,8 +156,46 @@ func (t *tikvHandlerTool) getRegionIDByKey(encodedKey []byte) (uint64, error) {
...
+		handleBytes, err = codec.EncodeKey(sc, nil, pkDts...)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		handle, err = kv.NewCommonHandle(handleBytes)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+	}
+	encodedKey := tb.RecordKey(handle)
```
Comment (xhebox) [inline server/http_handler.go:198]: `RecordKey` may be removed in https://github.com/pingcap/tidb/pull/22430

**Review by xhebox — COMMENTED** [review]
Rest LGTM

**Comment by coocood** [comment]
Forgot to update `tidb_http_api.md`?

**Comment by coocood** [comment]
LGTM

**Pull request merged**